spec/controllers/posts_controller_spec.rb

```diff
@@ -1,100 +1,151 @@
 require 'rails_helper'
 
 RSpec.describe PostsController, type: :controller do
-  let(:post) { FactoryBot.create(:post) }
+  let(:post_record) { FactoryBot.create(:post) }
+  let(:valid_attributes) { attributes_for(:post) }
+  let(:invalid_attributes) { { title: '' } }
+
+  shared_examples 'user do unauthorize action' do |action|
+    it 'redirects to root path' do
+      get action, params: { id: post_record.id } if %i[show edit].include?(action)
+      put action, params: { id: post_record.id } if %i[accept reject].include?(action)
+      delete action, params: { id: post_record.id } if action == :destroy
+
+      should redirect_to root_path
+      should set_flash[:alert]
+    end
+  end
 
-  describe 'Show' do
+  describe 'GET #show' do
     context 'user can not see un-accepted post of other user' do
       before { login_user }
 
-      it do
-        get :edit, params: { id: post.id }
-
-        should redirect_to root_path
-        should set_flash[:alert]
-      end
+      include_examples 'user do unauthorize action', :show
     end
   end
 
-  describe 'Edit' do
+  describe 'GET #edit' do
     context 'user can not edit post of other user' do
       before { login_user }
 
-      it do
-        get :edit, params: { id: post.id }
+      include_examples 'user do unauthorize action', :edit
+    end
+  end
+
+  describe 'POST #create' do
+    before { login_user }
+
+    context 'with valid params' do
+      it 'creates a new post' do
+        expect do
+          post :create, params: { post: valid_attributes }
+        end.to change(Post, :count).by(1)
+      end
+
+      it 'redirects to the created post' do
+        post :create, params: { post: valid_attributes }
+        should redirect_to Post.last
+      end
+    end
+
+    context 'with invalid params' do
+      it 'does not create a new post' do
+        expect do
+          post :create, params: { post: invalid_attributes }
+        end.not_to change(Post, :count)
+      end
 
-        should redirect_to root_path
-        should set_flash[:alert]
+      it "render 'new' template" do
+        post :create, params: { post: invalid_attributes }
+        should render_template 'new'
       end
     end
   end
 
-  describe 'Accept' do
-    context 'user are not authorize to accept post' do
-      before { login_user }
+  describe 'PUT #update' do
+    before { login_user }
+
+    let(:post_record) { FactoryBot.create(:post, user: controller.current_user) }
+
+    context 'with valid params' do
+      let(:new_attributes) { { title: 'new title' } }
 
-      it do
-        put :accept, params: { id: post.id }
+      it 'updates the post' do
+        put :update, params: { id: post_record.id, post: new_attributes }
```

Comment:
the `put :update, params: { id: post_record.id, post: new_attributes }` can refactored by wrap up and put after the context for example
```
describe 'PUT #update' do 
  let(:params) do
    {
    ///
    }
  end
context 'with valid params' do
  before do
    put :update, params: params
  end
  
  it 'updates the post' do
  //
  end
end

context '....'

end
```